cmd/tailscale/main.go

```diff
@@ -425,6 +473,31 @@ func (a *App) runBackend() error {
 			}
 		case e := <-backendEvents:
 			switch e := e.(type) {
+			case BugEvent:
+				backendLogID, _ := a.logIDPublicAtomic.Load().(string)
+				fallbackLog := fmt.Sprintf("BUG-%v-%v-%v", backendLogID, time.Now().UTC().Format("20060102150405Z"), randHex(8))
+
+				req, err := http.NewRequest("POST", "/localapi/v0/bugreport", nil)
+				if err != nil {
+					log.Printf("error creating new request for bug report: %v", err)
+					return err
+				}
+				w := newLocalApiResponseWriter()
+				h.ServeHTTP(w, req)
+				if w.StatusCode() > 300 {
+					log.Printf("bug report bad http status: %v", w.StatusCode())
+					a.bugReport <- fallbackLog
```

Comment:
I see that we read from a.bugReport in response to a UI `BugEvent`. Is it ever possible that we'll submit multiple bugs to the `a.bugReport` channel without getting corresponding UI `BugEvent`s to read them? If so, this could end up blocking because the channel is full. Should we have a timeout or something here and drop the log if we can't write to the channel?

Reply:
I don't think that is possible - we only get a backend BugEvent when there is a UI BugEvent, and when there's a UI BugEvent we always clear the channel before sending the backend BugEvent